### PR TITLE
🚼 Don't allow creation of child orgs within child orgs

### PR DESCRIPTION
**Pull request by rowanseymour**
And fix not setting date format correctly

### CODE REVIEW

File: temba/orgs/views.py [hunk 1]
```diff
@@ -3305,14 +3301,16 @@ class Workspace(SpaMixin, FormaxMixin, ContentMenuMixin, InferOrgMixin, OrgPerms
...
+
             menu.add_link(_("New Channel"), reverse("channels.channel_claim"), as_button=True)
 
             if self.has_org_perm("classifiers.classifier_connect"):
                 menu.add_link(_("New Classifier"), reverse("classifiers.classifier_connect"))
             if self.has_org_perm("tickets.ticketer_connect") and "ticketers" in settings.FEATURES:
                 menu.add_link(_("New Ticketing Service"), reverse("tickets.ticketer_connect"))
-            if self.has_org_perm("orgs.org_create_sub_org") and self.request.org.is_multi_org:
-                menu.add_modax(_("New Workspace"), "new-workspace", reverse("orgs.org_create_sub_org"))
+            if self.has_org_perm("orgs.org_create_child") and obj.is_multi_org and not obj.parent:
```
Comment (norkans7) [inline temba/orgs/views.py:3312]: In addition to this we need to probably also make sure we check the same to allow the view to be accessed by url

so make sure we do not allow a the view to be accessed unless current workspace obj `obj.is_multi_org and not obj.parent`

Comment (rowanseymour) [reply temba/orgs/views.py:3312]: there's an assert in Org.create_child so even if a user tries to go to the url it won't work

Comment (rowanseymour) [reply temba/orgs/views.py:3312]: but I guess a permission check would be cleaner...

Comment (norkans7) [reply temba/orgs/views.py:3312]: currently that would still not create the child workspace by the assert failing, probably that is also enough

File: temba/orgs/views.py [hunk 2]
```diff
@@ -2648,30 +2648,35 @@ class CreateOrgForm(forms.ModelForm):
...
-            initial["timezone"] = parent.timezone
-            initial["date_format"] = parent.date_format
+            initial["timezone"] = self.request.org.timezone
+            initial["date_format"] = self.request.org.date_format
             return initial
 
         def form_valid(self, form):
-            self.object = form.save(commit=False)
-            parent = self.org
-            child = parent.create_sub_org(self.object.name, self.object.timezone, self.request.user)
```
Comment (rowanseymour) [inline temba/orgs/views.py:2669]: weirdly we never set date_format on the new child org even tho it's on the form

File: temba/orgs/views.py (hunk 1 above)
Comment (rowanseymour) [reply temba/orgs/views.py:3321]: added a pre_process redirect

**Pull request merged**